cli/cmd/generate_aws_eks_audit.go

```diff
@@ -0,0 +1,917 @@
+	generateAwsEksAuditTfCommand.PersistentFlags().BoolVar(
+		&GenerateAwsEksAuditCommandState.BucketForceDestroy,
+		"enable_force_destroy",
+		false,
+		"enable force destroy s3 bucket")
+	generateAwsEksAuditTfCommand.PersistentFlags().IntVar(
+		&GenerateAwsEksAuditCommandState.BucketLifecycleExpirationDays,
+		"bucket_lifecycle_exp_days",
+		0,
+		"specify the s3 bucket lifecycle expiration days")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.BucketSseAlgorithm,
+		"bucket_sse_algorithm",
+		"",
+		"specify the encryption algorithm to use for S3 bucket server-side encryption")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.BucketSseKeyArn,
+		"bucket_sse_key_arn",
+		"",
+		"specify the kms key arn to be used for s3. (required when bucket_sse_algorithm is aws:kms & using an existing kms key)")
+	generateAwsEksAuditTfCommand.PersistentFlags().BoolVar(
+		&GenerateAwsEksAuditCommandState.BucketVersioning,
+		"enable_bucket_versioning",
+		true,
+		"enable s3 bucket versioning")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.EksAuditIntegrationName,
+		"integration_name",
+		"",
+		"specify the name of the eks audit integration")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.ExistingCloudWatchIamRoleArn,
+		"existing_cw_iam_role_arn",
+		"",
+		"specify existing cloudwatch iam role arn to use")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditExistingRoleState.Arn,
+		"existing_ca_iam_role_arn",
+		"",
+		"specify existing cross account iam role arn to use")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditExistingRoleState.ExternalId,
+		"existing_ca_iam_role_external_id",
+		"",
+		"specify existing cross account iam role external_id to use")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.ExistingFirehoseIamRoleArn,
+		"existing_firehose_iam_role_arn",
+		"",
+		"specify existing firehose iam role arn to use")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.FilterPattern,
+		"custom_filter_pattern",
+		"",
+		"specify a custom cloudwatch log filter pattern")
+	generateAwsEksAuditTfCommand.PersistentFlags().BoolVar(
+		&GenerateAwsEksAuditCommandState.FirehoseEncryptionEnabled,
+		"enable_firehose_encryption",
+		true,
+		"enable firehose encryption")
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.FirehoseEncryptionKeyArn,
+		"firehose_encryption_key_arn",
+		"",
+		"specify the kms key arn to be used with the Firehose")
+	generateAwsEksAuditTfCommand.PersistentFlags().IntVar(
+		&GenerateAwsEksAuditCommandState.KmsKeyDeletionDays,
+		"kms_key_deletion_days",
+		0,
+		"specify the waiting period, specified in number of days")
```

Comment:
is "waiting period" right here, or could it be expanded? Based on the flag name, sounds like it's the key deletion period.

Reply:
Updated to `specify the kms waiting period before deletion, in number of days`